pkg/controller/spd/indicator-plugin/resource_portrait_indicator_plugin_config_manager.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2022 The Katalyst Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package indicator_plugin
+
+import (
+	"encoding/json"
+	"strings"
+	"time"
+
+	apiworkload "github.com/kubewharf/katalyst-api/pkg/apis/workload/v1alpha1"
+	"github.com/kubewharf/katalyst-core/pkg/util/general"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	defaultTimeWindowInput      = 60
+	defaultTimeWindowOutput     = 60
+	defaultTimeWindowAggregator = apiworkload.Avg
+
+	defaultMethod       = ResourcePortraitMethodPredict
+	defaultCallInterval = 12 * 3600
+)
+
+type ResourcePortraitConfigManager interface {
+	Refresh(ns, name string) bool
+	Filter(runtime.Object) []string
+	Find(scope string) *ResourcePortraitConfig
+}
+
+type resourcePortraitConfigManagerImpl struct {
+	cmLister     corelisters.ConfigMapLister
+	scopeConfigs map[string]*ResourcePortraitConfig
+}
+
+// ResourcePortraitConfig is the config for one scope, different scopes need their own config.
+type ResourcePortraitConfig struct {
```

Comment:
The definition of `ResourcePortraitConfig` can be relocated to [katalyst-api](https://github.com/kubewharf/katalyst-api/tree/main/pkg/apis/config/v1alpha1), and the extendedIndicator struct can also be defined there, similar to `TransparentMemoryOffloadingIndicators` in [pkg/apis/config/v1alpha1/tmo.go](https://github.com/kubewharf/katalyst-api/blob/main/pkg/apis/config/v1alpha1/tmo.go)